crypto/keyring/output.go

```diff
@@ -1,106 +1,73 @@
 package keyring
 
 import (
+	"github.com/cosmos/cosmos-sdk/codec"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+// TODO: Move this file to client/keys
+// Use protobuf interface marshaler rather then generic JSON
+
 // KeyOutput defines a structure wrapping around an Info object used for output
 // functionality.
 type KeyOutput struct {
-	Name      string                 `json:"name" yaml:"name"`
-	Type      string                 `json:"type" yaml:"type"`
-	Address   string                 `json:"address" yaml:"address"`
-	PubKey    string                 `json:"pubkey" yaml:"pubkey"`
-	Mnemonic  string                 `json:"mnemonic,omitempty" yaml:"mnemonic"`
-	Threshold uint                   `json:"threshold,omitempty" yaml:"threshold"`
-	PubKeys   []multisigPubKeyOutput `json:"pubkeys,omitempty" yaml:"pubkeys"`
+	Name     string `json:"name" yaml:"name"`
+	Type     string `json:"type" yaml:"type"`
+	Address  string `json:"address" yaml:"address"`
+	PubKey   string `json:"pubkey" yaml:"pubkey"`
```

Comment:
```suggestion
	PubKey   *Any `json:"pubkey" yaml:"pubkey"`
```

related to https://github.com/cosmos/cosmos-sdk/pull/7477#pullrequestreview-608323873, I would also prefer `Any` here, so that 1. the key type is shown, and 2. users can copy-paste

Reply:
> We could update the format - but that will require passing a codec down to the keyring functions.

We don't need, we just need to do `pkAny := codec.NewAnyWithValue(pk)` in the `NewKeyOutput()` function

Reply:
`KeyOutput` is meant to be used for direct printing (struct fields printing) or printing use json/yaml serialization. Calling `MarshalJSON` for type *types.Any is not supported and fails.

Reply:
Ah, good point, let's keep `PubKey string` then

Reply:
> let's keep PubKey string then, but the string is the JSON repr of the Any.

Yes. 
The `KeyOutput` can be used with any output serialization, including yaml (eg `printTextInfos` function)